lti_tool/views.py

```diff
@@ -0,0 +1,317 @@
+
+    def get(self, request, *args, **kwargs):
+        # Edit links to point to LTI views
+        module_model_objs = CourseModule.objects.filter(course_instance=self.instance)
+        learningobjects = LearningObject.objects.filter(course_module__course_instance=self.instance)
+        for module in self.content.data['modules']:
+            module_model_obj = next((x for x in module_model_objs if x.id == module['id']), None)
+            module.update({'link': module_model_obj.get_url("lti-module")})
+            for exercise in module['children']:
+                lo_model_obj = next((x for x in learningobjects if x.id == exercise['id']), None)
+                exercise.update({'link': lo_model_obj.get_url("lti-exercise")})
+        return super().get(request, *args, **kwargs)
+
+
+class LtiModuleView(ModuleView, LtiSessionMixin):
+
+    access_mode = ACCESS.ENROLLED
+    template_name = "lti_tool/lti_module.html"
+
+    def get(self, request, *args, **kwargs):
+        learningobjects = LearningObject.objects.filter(course_module=self.module)
+        learningobjects_dict = { obj.id: obj for obj in learningobjects }
+        # Can't use self.children for iteration, so this instead
+        flat_module = self.content.flat_module(self.module)
+        exercises = [entry for entry in flat_module if entry['type'] == 'exercise']
+        for exercise in exercises:
+            learningobj = learningobjects_dict[exercise['id']]
+            exercise['link'] = learningobj.get_url('lti-exercise')
+        return super().get(request, *args, **kwargs)
+
+
+class LtiExerciseView(ExerciseView, LtiSessionMixin):
+
+    access_mode = ACCESS.ENROLLED
+    template_name = "lti_tool/lti_exercise.html"
+    post_url_name = "lti-exercise"
+    submission_url_name = "lti-submission"
+    exercise_url_name = "lti-exercise"
+
+    def post(self, request, *args, **kwargs):
+        return super().post(request, *args, **kwargs, redirect_view='lti-submission')
+
+
+class LtiSubmissionView(SubmissionView, LtiSessionMixin):
+
+    access_mode = ACCESS.ENROLLED
+    template_name = "lti_tool/lti_submission.html"
+    submission_url_name = "lti-submission"
+    exercise_url_name = 'lti-exercise'
+
+
+class LtiSelectContentMixin(LtiSessionMixin):
+
+    access_mode = ACCESS.TEACHER
+
+    def get_resource_objects(self):
+        super().get_resource_objects()
+        self.is_teacher = Enrollment.objects.filter(
+            user_profile=self.request.user.userprofile,
+            role=Enrollment.ENROLLMENT_ROLE.TEACHER,
+            status=Enrollment.ENROLLMENT_STATUS.ACTIVE,
+        ).exists()
+
+    def get_common_objects(self):
+        super().get_common_objects()
+        self.parse_lti_session_params()
+        self.teacher_enrollments = Enrollment.objects.select_related("course_instance").filter(
+            user_profile=self.request.user.userprofile,
+            role=Enrollment.ENROLLMENT_ROLE.TEACHER
+        )
```

Comment:
```suggestion
        self.teacher_enrollments = Enrollment.objects.select_related("course_instance").filter(
            user_profile=self.request.user.userprofile,
            role=Enrollment.ENROLLMENT_ROLE.TEACHER,
            status=Enrollment.ENROLLMENT_STATUS.ACTIVE,
        )
```
Must check that the status is ACTIVE.